src/poprox_recommender/nrms.py

```diff
@@ -172,29 +162,24 @@ def generate_recommendations(model, articles, article_vectors, user_embeddings,
 
 
 def select_with_model(
-    todays_articles,
-    todays_article_features,
+    article_similarity_matrix,
     past_article_features,
     clicked_articles,
-    model_checkpoint,
+    model,
     model_device,
     num_slots=10,
     max_clicks_per_user=10,
 ):
     # Translate clicks JSON to dataframe
     user_df = build_clicks_df(clicked_articles)
 
-    # Load model and build embedding tables
-    model = load_model(model_checkpoint, model_device)
+    # Build embedding tables
     past_article_embeddings, _ = build_article_embeddings(past_article_features, model, model_device)
     user_embeddings = build_user_embeddings(user_df, past_article_embeddings, model, model_device, max_clicks_per_user)
 
-    _, todays_article_vectors = build_article_embeddings(todays_article_features, model, model_device)
-
     recommendations = generate_recommendations(
         model,
-        todays_articles,
-        todays_article_vectors,
```

Comment:
It looks like `generate_recommendations` still uses `articles` and `article_vectors` so we should keep these

Reply:
Thanks for catching that! I know there must be something missing 😅Hopefully when we have more test cases built up it'll be easier to catch!

Reply:
💯 

Every change is like walking a high wire right now, and changing that should be a high priority. Talked to @mdekstrand about that a bit this afternoon, and we were thinking one of the things this repo should have is a script that downloads the model and some data, runs them through the recommender, and just prints out the results. It's not really a substitute for a test suite, but it would give us a way to double-check that things are working the way we think they should by 👀 ing the output

Reply:
Totally agree. For the model / test data downloading part I assume we might need to integrate the S3 logic? Happy to help with that if needed (or if it's already in your next week's plan then I can wait until you create a PR and add more stuff to that).

Reply:
We can probably just have DVC pull directly from S3: https://dvc.org/doc/command-reference/import-url

Then we can load from that local file (and speed up iteration).

Reply:
(Adding DVC to this repo is a "do this very soon" task, and I was going to connect with @aisa6148 on Monday about kicking that off.)